flocker/node/agents/testtools/_blockdevice.py

```diff
@@ -891,117 +891,81 @@ class InvalidConfig(Exception):
     """
 
 
-# Highly duplicative of other constants.  FLOC-2584.
-class ProviderType(Names):
-    """
-    Kinds of compute/storage cloud providers for which this module is able to
-    build ``IBlockDeviceAPI`` providers.
-    """
-    openstack = NamedConstant()
-    aws = NamedConstant()
-    gce = NamedConstant()
-    rackspace = NamedConstant()
+# XXX Remember to comment on or close FLOC-2584.
 
 
-def get_blockdeviceapi(provider_type):
+def get_blockdeviceapi():
     """
     Validate and load cloud provider's yml config file.
     Default to ``~/acceptance.yml`` in the current user home directory, since
     that's where buildbot puts its acceptance test credentials file.
     """
-    config = get_blockdevice_config(provider_type)
-    provider = _provider_for_provider_type(provider_type)
+    config = get_blockdevice_config()
+    backend_name = config.pop('backend')
+    provider = Providers.lookupByName(backend_name.upper())
     factory = _BLOCKDEVICE_TYPES[provider]
```

Comment:
This still won't work for external drivers, correct?

Just a passing observation, obviously not worse than what was here.